lib/OpenCloud/ObjectStore/Resource/DataObject.php

```diff
@@ -354,6 +382,63 @@ public function delete($params = array())
     {
         return $this->getService()->getClient()->delete($this->getUrl())->send();
     }
+    
+    /**
+     * @param string $destination Path (`container/object') of other object to symlink this object to
+     * @return null|\Guzzle\Http\Message\Response The response or null if $this is not empty
+     */
+    public function createSymlinkTo($destination)
+    {
+        if (!$this->name) {
+            throw new Exceptions\NoNameError(Lang::translate('Object has no name'));
+        }
+
+        if ($this->getContentLength() == 0) {
+            $response = $this->getService()
+                ->getClient()
+                ->createRequest('PUT', $this->getUrl(), array(
+                    HeaderConst::X_OBJECT_MANIFEST => (string) $destination
+                ))
+                ->send();
+
+            if ($response->getStatusCode() == 201) {
+                $this->setManifest($source);
+            }
+
+            return $response;
+        }
+
+        return null;
+    }
+
+    /**
+     * @param string $source Path (`container/object') of other object to symlink this object from
+     * @return null|DataObject The symlinked object or null if object already exists and is not empty
+     */
+    public function createSymlinkFrom($source)
+    {
+        if (!strlen($source)) {
+            throw new Exceptions\NoNameError(Lang::translate('Object has no name'));
+        }
+
+        // Use ltrim to remove leading slash from source
+        list($containerName, $resourceName) = explode("/", ltrim($source, '/'), 2);
+
+        $container = $this->getService()->getContainer($containerName);
+
```

Comment:
See [my comment above on throwing exceptions instead of returning null](https://github.com/rackspace/php-opencloud/pull/565/files#r26799129). If you use exceptions, you could refactor the code below to use guard clauses as shown below, which tend to be more readable in my opinion:

```php
if ($container->objectExists($resourceName)) {
    throw new ... // appropriate exception
}

$object = $container->getPartialObject($source);
if ($object->getContentLength() > ) {
    throw new ... // appropriate exception
}

return $container->uploadObject($resourceName, 'data', array(
    HeaderConst::X_OBJECT_MANIFEST => (string) $this->getUrl()
));
```